src/vmm/benches/block_request.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Benchmarking cases:
+//   * `Queue.pop`
+//   * `Queue.add_used`
+//   * `DescriptorChain.next_descriptor`
+
+use criterion::{criterion_group, criterion_main, Criterion};
+use vm_memory::GuestAddress;
+use vmm::devices::virtio::block::virtio::test_utils::RequestDescriptorChain;
+use vmm::devices::virtio::block::virtio::{Request, RequestHeader, VIRTIO_BLK_T_IN};
+use vmm::devices::virtio::test_utils::VirtQueue;
+use vmm::utilities::test_utils::single_region_mem;
+
+pub fn block_request_benchmark(c: &mut Criterion) {
+    let mem = single_region_mem(2 * 65562);
+    let virt_queue = VirtQueue::new(GuestAddress(0), &mem, 256);
+
+    // We don't really care about what request is. We just
+    // need it to be valid.
+    let chain = RequestDescriptorChain::new(&virt_queue);
+    let request_header = RequestHeader::new(VIRTIO_BLK_T_IN, 99);
+    chain.set_header(request_header);
+
+    let mut queue = virt_queue.create_queue();
+
+    // SAFETY: queue has 1 desc chain
+    let desc = unsafe { queue.pop(&mem).unwrap_unchecked() };
+
+    c.bench_function("request_parse", |b| {
+        b.iter(|| {
+            let desc = std::hint::black_box(&desc);
+            // SAFETY: will never fail
+            unsafe { Request::parse(desc, &mem, 1024).unwrap_unchecked() };
```

Comment:
just ignore the return value here as well

Reply:
done